### ISSUE TITLE
Allow callbacks on client connection failure

### ISSUE THREAD

**Issue by maxfischer2781** [xrootd/xrootd#1388]
In order to track/monitor connection failures on clients, it would be great to allow client plugins to run jobs when a connection fails. Currently, plugins seem restricted to handlers for connection success via ``PostMaster::SetOnConnectHandler``.

Either having the option to run the OnConnectHandler on any connection result, or a separate handler for failing connection should be sufficient.

**Comment by simonmichal** [comment]
@maxfischer2781 : I assigned this to `5.2.0` as `5.1.0` will be the last one that will be released for `SLC6`, meaning that afterwards we will have full C++11 (event C++14) support and I'll be able then to give you much cleaner API that takes a lambda as a callback.

**Comment by simonmichal** [comment]
@maxfischer2781 : it is implemented by c5e76b10a1ffc701cfa4171cf6db8602c3e573e4

You should be able to set a callback using following syntax:

```
  XrdCl::DefaultEnv::GetPostMaster()->SetConnectionErrorHandler(
      []( const URL &url, const XRootDStatus &status )
      {
        std::cout << "calling connection error handler" << std::endl;
      } );
```

Let me know if we can close this issue or is there anything else that is needed?

**Comment by simonmichal** [comment]
@maxfischer2781 : I suppose we can close this ticket?
